AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/AwsCryptographyKeyStoreAdminOperations.dfy

```diff
@@ -39,11 +38,51 @@ module AwsCryptographyKeyStoreAdminOperations refines AbstractAwsCryptographyKey
     config.storage.Modifies + MutationLie()
   }
 
-  method resolveKmsStratgety(
+  method ProvideKMSClient(
+    kmsClient?: Option<KMS.Types.IKMSClient> := None
+  )
+    returns (output: Result<KMS.Types.IKMSClient, KMS.Types.Error>)
+    // Because Dafny is not able to parse
+    // the code that Smithy-Dafny produces for reference types inside a union,
+    // the requires kms.ValidState() and modifies kmsClient are commented out.
+    // requires kms.kmsClient.Some? ==> kms.kmsClient.value.ValidState()
+    // modifies (if kms.kmsClient.Some? then kms.kmsClient.value.Modifies else {})
```

Comment:
can we add a tracking GHI to the dafny team about this?
I am not enthused about axiom here

Reply:
@seebees Did you cut a GHI or Ticket for the Union issue?

Reply:
Resolved in a2a28ea23